### PR TITLE
feat: Add ip string validator

### LINKED ISSUE #11: Add `l.string().ip()` - Validates both IPv4 and IPv6

### PR DESCRIPTION
**Pull request by dipinsonwani**
Add ip string validator
Fix #11

### CODE REVIEW

File: packages/luthor/lib/src/validators/string_validator.dart [hunk 1]
```diff
@@ -82,6 +83,13 @@ class StringValidator extends Validator {
     return this;
   }
 
+  /// Validates that the string is an ip with optional parameter for IPv4 and IPv6 as "v4" and "v6" respectively.
+  StringValidator ip({String? message, String? version}) {
```
Comment (exaby73) [inline packages/luthor/lib/src/validators/string_validator.dart:87]: Can you make this an enum value? For example: `IpVersion.v4`. I'd like not to have magic strings and have everything as typesafe as possible

**Review by exaby73 — CHANGES_REQUESTED** [review]

File: packages/luthor/lib/src/validators/string_validator.dart (hunk 1 above)
Comment (dipinsonwani) [reply packages/luthor/lib/src/validators/string_validator.dart:87]: sure, no issue

**Comment by exaby73** [comment]
Can you add tests for this?

**Comment by dipinsonwani** [comment]
sure

File: packages/luthor/lib/src/annotations/validators/ip.dart [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class IsIp {
+  final String? version;
```
Comment (exaby73) [inline packages/luthor/lib/src/annotations/validators/ip.dart:2]: You can use the `IpVersion` enum here as well

File: packages/luthor/lib/src/validators/string_validator.dart [hunk 3]
```diff
@@ -82,6 +84,13 @@ class StringValidator extends Validator {
     return this;
   }
 
+  /// Validates that the string is an ip with optional parameter for IPv4 and IPv6 as "v4" and "v6" respectively.
+  StringValidator ip({String? message, IpVersion? version}) {
+    validations
+        .add(StringIpValidation(customMessage: message, version: version));
```
Comment (exaby73) [inline packages/luthor/lib/src/validators/string_validator.dart:90]: Add a comma here and reformat please

File: packages/luthor_generator/example/lib/another_sample.dart [hunk 4]
```diff
@@ -12,6 +12,7 @@ class AnotherSample with _$AnotherSample {
     required int id,
     @JsonKey(name: 'full_name') String? name,
     @IsEmail(message: "Invalid email") required String email,
+    @IsIp(version: "v4") String? ip,
```
Comment (exaby73) [inline packages/luthor_generator/example/lib/another_sample.dart:15]: Once the `IpVersion` enum is used, this should be updated as well

File: packages/luthor_generator/lib/helpers/validations/string_validations.dart [hunk 5]
```diff
@@ -181,3 +182,29 @@ void _checkAndWriteContainsValidation(
...
+
+void _checkAndWriteIpValidation(
+  StringBuffer buffer,
+  ParameterElement param,
+) {
+  final ipAnnotation = getAnnotation(isIpChecker, param);
+  if (ipAnnotation != null) {
+    buffer.write('.ip(');
+
+    final version = ipAnnotation.getField('version')?.toStringValue();
```
Comment (exaby73) [inline packages/luthor_generator/lib/helpers/validations/string_validations.dart:194]: Enum would be updated here as well

**Review by exaby73 — CHANGES_REQUESTED** [review]

**Review by exaby73 — APPROVED** [review]

**Pull request merged**